src/TypeOps.hs

```diff
@@ -46,3 +47,12 @@ import           Control.Type.Operator (type ($), type (<+>))
 type family Each (c :: [k -> Constraint]) (as :: [k]) where
     Each c '[] = (() :: Constraint)
     Each c (h ': t) = (c <+> h, Each c t)
+
+-- | Map several constraints over a single variable.
+--
+-- @
+-- a :: With [Show, Read] a => a -> a
+-- =
+-- a :: (Show a, Read a) => a -> a
+-- @
+type With a b = a <+> b
```

Comment:
Perhaps it's even more weird to first write `Each` via `<+>`, then define `With` via `Each` :smile: 
I can write in haddock comment that `With a b` is equal to `Each a [b]`